apps/web/vibes/soul/primitives/breadcrumbs/index.tsx

```diff
@@ -1,20 +1,31 @@
 import { clsx } from 'clsx';
 import { ChevronRight } from 'lucide-react';
+import { ComponentPropsWithoutRef } from 'react';
 
 import { Stream, Streamable } from '@/vibes/soul/lib/streamable';
 import { AnimatedLink } from '@/vibes/soul/primitives/animated-link';
 
-export interface Breadcrumb {
-  label: string;
-  href: string;
-}
-
-export interface BreadcrumbsProps {
-  breadcrumbs: Streamable<Breadcrumb[]>;
+/**
+ * This component supports various CSS variables for theming. Here's a comprehensive list, along
+ * with their default values:
+ *
+ * ```css
+ * :root {
+ *   --breadcrumbs-font-family: var(--font-family-body);
+ *   --breadcrumbs-primary-text: hsl(var(--foreground));
+ *   --breadcrumbs-secondary-text: hsl(var(--contrast-500));
+ *   --breadcrumbs-icon: hsl(var(--contrast-500));
+ *   --breadcrumbs-hover: hsl(var(--primary));
+ * }
+ * ```
+ */
+export function Breadcrumbs({
+  breadcrumbs: streamableBreadcrumbs,
+  className,
+}: {
+  breadcrumbs: Streamable<Array<ComponentPropsWithoutRef<typeof AnimatedLink> & { id: string }>>;
```

Comment:
I _really_ dislike doing this here as well. The API surface area of the breadcrumbs should be even more limited than `AnimatedLink`. I think we need to define what a "breadcrumb item" is, completely independently of `AnimatedLink`. `AnimatedLink` is an implementation detail and here we're coupling it with the breadcrumb items